micromamba/tests/test_env.py

```diff
@@ -200,3 +201,64 @@ def test_explicit_export_topologically_sorted(tmp_home, tmp_prefix):
     assert indices["libzlib"] < indices["python"]
     assert indices["python"] < indices["pip"]
     assert indices["python"] < indices["jupyterlab"]
+
+
+@pytest.mark.parametrize("shared_pkgs_dirs", [True], indirect=True)
+def test_env_create_pypi(tmp_home, tmp_root_prefix, tmp_path):
+    env_prefix = tmp_path / "myenv"
+
+    create_spec_file = tmp_path / "env-pypi-pkg-test.yaml"
+
+    shutil.copyfile(__this_dir__ / "env-pypi-pkg-test.yaml", create_spec_file)
+
+    res = helpers.run_env("create", "-p", env_prefix, "-f", create_spec_file, "-y", "--json")
+    assert res["success"]
+
+    # Check that pypi-pkg-test is installed using pip for now
+    # See: https://github.com/mamba-org/mamba/issues/2059
+    pip_list_output = helpers.umamba_run("-p", env_prefix, "pip", "list", "--format=json")
+    pip_packages_list = yaml.safe_load(pip_list_output)
+    assert any(pkg["name"] == "pypi-pkg-test" for pkg in pip_packages_list)
+
+
+@pytest.mark.parametrize("shared_pkgs_dirs", [True], indirect=True)
+def test_env_create_update_pypi(tmp_home, tmp_root_prefix, tmp_path):
+    env_prefix = tmp_path / "myenv"
+    env_prefix_conda = tmp_path / "myenv-conda"
+
+    create_spec_file = tmp_path / "env-requires-pip-install.yaml"
+    update_spec_file = tmp_path / "env-pypi-pkg-test.yaml"
+
+    shutil.copyfile(__this_dir__ / "env-requires-pip-install.yaml", create_spec_file)
+    shutil.copyfile(__this_dir__ / "env-pypi-pkg-test.yaml", update_spec_file)
+
+    res = helpers.run_env("create", "-p", env_prefix, "-f", create_spec_file, "-y", "--json")
+    assert res["success"]
+
+    # Check that pypi-pkg-test is not installed before the update
+    pip_list_output = helpers.umamba_run("-p", env_prefix, "pip", "list", "--format=json")
+    pip_packages_list = yaml.safe_load(pip_list_output)
+    assert all(pkg["name"] != "pypi-pkg-test" for pkg in pip_packages_list)
+
+    umamba_packages_list = helpers.umamba_list("-p", env_prefix, "--json")
+    assert all(pkg["name"] != "pypi-pkg-test" for pkg in umamba_packages_list)
+
+    res = helpers.run_env("update", "-p", env_prefix, "-f", update_spec_file, "-y", "--json")
+    assert res["success"]
+
+    # Check that pypi-pkg-test (that is part of the update environment) is installed after the update
+    # See: https://github.com/mamba-org/mamba/issues/2059
+    pip_list_output = helpers.umamba_run("-p", env_prefix, "pip", "list", "--format=json")
+    packages_list = yaml.safe_load(pip_list_output)
+    assert any(pkg["name"] == "pypi-pkg-test" for pkg in packages_list)
+
+    # Create a new environment with the same spec file but with conda within the shell
+    subprocess.run(f"conda update -p {update_spec_file} -f {create_spec_file} -y")
```

Comment:
I think using `conda` (even for tests) is a very bad idea. We can just know how it should behave and check the result.
This also looks like mixing updates with `conda` and `mamba` and it shouldn't.
The error in the CI seems to be caused by setting the target prefix as a yaml file` -p {update_spec_file}` ?